glean-core/src/ping/mod.rs

```diff
@@ -41,6 +42,21 @@ impl Default for PingMaker {
     }
 }
 
+// The string fields that are required in a client_info section.
+// Must match the fields defined in the pipeline schema:
+//     https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/5b34499d19658b1a0926a68f55d12e09ad022c60/templates/include/glean/glean.1.schema.json#L35
+// When these keys are missing from the database, they are filled in with the value "missing".
+static REQUIRED_CLIENT_INFO_FIELDS: Lazy<Vec<String>> = Lazy::new(|| {
+    vec![
+        "app_build".into(),
+        "app_display_version".into(),
+        "architecture".into(),
+        "first_run_date".into(),
+        "os".into(),
+        "os_version".into(),
+    ]
+});
```

Comment:
No need to allocate (we will need to allocate later on insert though):

```suggestion
static REQUIRED_CLIENT_INFO_FIELDS: &[&str] = &[
    "app_build",
    "app_display_version",
    "architecture",
    "first_run_date",
    "os",
    "os_version",
];
```